src/buffer/out/AttrRowIterator.cpp

```diff
@@ -84,11 +89,13 @@ AttrRowIterator AttrRowIterator::operator--(int)
 
 const TextAttribute* AttrRowIterator::operator->() const
 {
+    THROW_HR_IF(E_BOUNDS, _exceeded);

```

Comment:
```suggestion
    // This should be technically impossible as the movement stops this condition.
    FAIL_FAST_IF(_exceeded);
```

Reply:
Resolving because it makes more sense to have it be flexible and let the `E_BOUNDS` be caught/handled elsewhere.